static/app/components/searchSyntax/utils.tsx

```diff
@@ -252,27 +250,59 @@ export const getKeyName = (
   >,
   options: GetKeyNameOpts = {}
 ) => {
-  const {aggregateWithArgs, showExplicitTagPrefix = false} = options;
+  const {aggregateWithArgs} = options;
   switch (key.type) {
     case Token.KEY_SIMPLE:
       return key.value;
     case Token.KEY_EXPLICIT_TAG:
-      if (showExplicitTagPrefix) {
-        return key.text;
-      }
       return key.key.value;
     case Token.KEY_AGGREGATE:
       return aggregateWithArgs
         ? `${key.name.value}(${key.args ? key.args.text : ''})`
         : key.name.value;
     case Token.KEY_EXPLICIT_NUMBER_TAG:
-      // number tags always need to be expressed with the
-      // explicit tag prefix + type
       return key.text;
     case Token.KEY_EXPLICIT_STRING_TAG:
-      if (showExplicitTagPrefix) {
-        return key.text;
-      }
+      return key.text;
+    case Token.KEY_EXPLICIT_FLAG:
+      return key.text;
+    case Token.KEY_EXPLICIT_NUMBER_FLAG:
+      return key.text;
+    case Token.KEY_EXPLICIT_STRING_FLAG:
+      return key.text;
+    default:
+      return '';
+  }
+};
+
+/**
+ * Utility to get the public facing label of any type of key.
+ * Used to format a key in a user friendly way. This value
+ * should only be used for display, and not passed to the API.
+ * For the value to use in the API, see `getKeyName`.
+ */
+export const getKeyLabel = (
```

Comment:
Glad to see these behaviors explicitly split up into two functions, this makes a lot of sense